go/toolchains.rst

```diff
@@ -18,22 +27,568 @@ customize the behavior of the core_ go rules.
++--------------------------------+-----------------------------+-----------------------------------+
+| This must be the same go toolchain object you got this function from.                            |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`sources`               | :type:`list<File>`          | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| An iterable of Go source files.                                                                  |
++--------------------------------+-----------------------------+-----------------------------------+
+
+library
+~~~~~~~
+
+This emits actions to compile Go code into an archive.
+It supports embedding, cgo dependencies, coverage, and assembling and packing .s files.
+
+It returns a tuple of GoLibrary and GoEmbed.
+
++--------------------------------+-----------------------------+-----------------------------------+
+| **Name**                       | **Type**                    | **Default value**                 |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`ctx`                   | :type:`string`              | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The current rule context, used to generate the actions.                                          |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`go_toolchain`          | :type:`the go toolchain`    | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| This must be the same go toolchain object you got this function from.                            |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`srcs`                  | :type:`list<File>`          | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| An iterable of go source Files to be compiled.                                                   |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`deps`                  | :type:`list<GoLibrary>`     | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The list of direct dependancies of this package.                                                 |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`cgo_info`              | :type:`CgoInfo`             | :value:`None`                     |
```

Comment:
I wonder if we should leave this undocumented? I'd love to get rid of this if we ever gain the capability to produce actions that compile and link C files.